server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4796,6 +4815,40 @@ private boolean localDirectoryContains(Directory localDirectory, String file, lo
         return false;
     }
 
+    private void beforeSegmentDownload(RemoteRefreshSegmentTracker downloadStatsTracker, long incomingFileSize) {
+        if (!indexSettings.isRemoteStoreEnabled()) {
+            return;
+        }
+        downloadStatsTracker.incrementTotalDownloadsStarted();
```

Comment:
This is used to track overall downloads and not file level. Lets fix this.

Reply:
`downloadStatsTracker.incrementTotalDownloadsStarted();` should be called only once per invocation of `syncSegmentsFromRemoteSegmentStore`.